tethysext/atcore/services/resource_condor_workflow.py

```diff
@@ -100,7 +100,9 @@ def prepare(self):
 
         # Set relationships
         for job, _ in user_defined_jobs:
-            update_resource_status.add_parent(job)
+            has_no_children = job.children_nodes.count() == 0
+            if has_no_children:
+                update_resource_status.add_parent(job)
```

Comment:
What is this change doing?
Tests are failing on this.

Reply:
This is the change that adds the finalize job only at the end of the workflow chain instead of to every node so that the diagram doesn't become messy. We are fixing the test now.